### DISCUSSION TITLE
The question about the nyul mask

### DISCUSSION THREAD

**Discussion by weifanliao** [jcreinhold/intensity-normalization#77] [Q&A] [answered]
Dear Sir, 
    I have a question to ask you in the process of image normalization. I have one set of images where the skull has been removed and another set of images where the tumor area of a patient with cancer (other normal tissue has been removed).
In this case, I want to apply the nyul transformation to the first group and the second group, respectively.My question is, can I use your nyul Python API directly?Use need not add mask in code?Because, in my attempts to change the        nyul_normalizer.fit(images)        to nyul_normalizer.fit(images,masks)  and   normalized = [new_nyul_normalizer(image) for image in images]     to    a = new_nyul_normalizer(new_images[i],new_masks[i])     and no error.    (new_nyul_normalizer(new_images[i],new_masks[i])  :What I want to express in this sentence is that in the conversion process, I added mask when there was only image).
    Thank you so much.
（I mentioned the      nyul Python API      is in the documentation of the code in the following pictures

![b10ca9d31259e95cf7a80786d437382](https://github.com/jcreinhold/intensity-normalization/assets/134304141/f450c784-2474-4bda-8633-f5c983177ea6)

 ）

                                                                                                                                                                                     weifanliao

**Comment by jcreinhold** [comment]
Sorry, but I'm not quite following what you're trying to accomplish based on the information you provided. Could you please try to rephrase your question? Perhaps provide the specific code you have a question about (removing hard-coded paths) and point to where you are running into unexpected behavior in that code snippet. See [this page](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) for information about how to format code in Markdown on GitHub.

In general, the Nyul intensity normalization method works when you have MR images of the same anatomy (e.g., a set of brain MR images or a set of knee MR images). You can't `fit` a `NyulNormalize` instance on a set of brain MR images and use the fit normalizer on images of MR tumors.

If you are trying to normalize two sets of images of different anatomical structures, you'd do best with a simpler normalization scheme like Z-score. Or, perhaps, you can `fit` two instances of `NyulNormalize`; one for the skull-stripped images and one for the tumor images. Hope that helps.

**Comment by jcreinhold** [comment, marked as answer]
If you're simply asking if you _need_ to provide masks, and your images are already skull-stripped/background stripped, then the answer is no. You don't _need_ to provide foreground masks (if your background is all exactly zero and the foreground is all strictly positive).

**Reply by jcreinhold** [reply]
Yes. You don't need to provide a mask if the background is zero and the foreground is positive.

**Reply by weifanliao** [reply]
Thank you, Sir. I just deleted my comment and am editing it again. Thanks again!
